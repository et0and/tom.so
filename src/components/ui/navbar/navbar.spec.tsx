import React from "react";
import { render } from "@testing-library/react";
import Navbar from "./navbar";

// Mock the next/link and next/navigation modules
jest.mock("next/link", () => {
  const MockedLink = ({
    children,
    href,
  }: {
    children: React.ReactNode;
    href: string;
  }) => <a href={href}>{children}</a>;
  MockedLink.displayName = "MockedLink";
  return MockedLink;
});

jest.mock("next/navigation", () => ({
  usePathname: jest.fn(),
}));

describe("<Navbar/>", () => {
  it("renders correctly", () => {
    const { usePathname } = require("next/navigation");
    usePathname.mockReturnValue("/");

    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });

  it("does not render when pathname starts with /keystatic", () => {
    const { usePathname } = require("next/navigation");
    usePathname.mockReturnValue("/keystatic/some-path");

    const { container } = render(<Navbar />);
    expect(container.firstChild).toBeNull();
  });
});
