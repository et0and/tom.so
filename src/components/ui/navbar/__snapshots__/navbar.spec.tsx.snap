// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> renders correctly 1`] = `
<div>
  <nav
    class="flex items-center tracking-tighter justify-between h-8 px-6 pt-6 dark:bg-stone-900"
  >
    <a
      href="/"
    >
      <h1
        class="md:block hidden"
      >
        Tom Hackshaw
      </h1>
      <div
        class="md:hidden block focus-visible:outline-offset-2 focus-visible:outline-blue-700 outline-transparent focus-visible:border-blue-700 border-transparent border-2 hover:underline hover:text-blue-700 dark:hover:text-teal-200 transition-colors duration-100 w-7 h-7 bg-black dark:bg-white rounded-full"
      >
        <h1
          class="sr-only"
        >
          Tom Hackshaw
        </h1>
      </div>
    </a>
    <div
      class="flex md:items-center space-x-4 text-lg"
    >
      <a
        href="/about"
      >
        About
      </a>
      <a
        href="/work"
      >
        Work
      </a>
      <a
        href="/posts"
      >
        Writing
      </a>
    </div>
  </nav>
</div>
`;
